.circleci/config.yml

```diff
@@ -9,6 +9,26 @@ jobs:
     - run:
         command: /usr/sbin/softwareupdate --install-rosetta --agree-to-license
         name: Installing Rosetta 2
+    - run:
+        command: |-
+          # install jdk
+          curl -Lf \
+           https://github.com/adoptium/temurin21-binaries/releases/download/jdk-21.0.5%2B11/OpenJDK21U-jdk_x64_mac_hotspot_21.0.5_11.tar.gz -o /tmp/jdk.tar.gz
+
+          mkdir $HOME/jdk \
+            && cd $HOME/jdk \
+            && cat /tmp/jdk.tar.gz | tar --strip-components=1 -xzC .
+
+          export JAVA_ROOT="/Library/Java/JavaVirtualMachines/jdk-21.jdk"
+          export JAVA_HOME="$JAVA_ROOT/Contents/Home"
+          export PATH="$JAVA_HOME/bin:$PATH"
+          mv $HOME/jdk $JAVA_ROOT
+          /usr/libexec/java_home -V
+          java -version
+          mkdir -p $HOME/.gradle
+          echo "org.gradle.java.home=$JAVA_HOME" >> $HOME/.gradle/gradle.properties
```

Comment:
Some notes:

- `JAVA_ROOT` vs `JAVA_HOME` accounts for the `../Contents/Home` path prepended to mac JDKs.
- `/jdk` is not writable on modern macOS, so the JDK is unpacked at `$HOME/jdk`
- I am not sure `export JAVA_HOME=...` will impact subsequent steps, so `org.gradle.java.home=...` is set instead